accepted/2022/version-selection.md

```diff
@@ -0,0 +1,449 @@
+dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
+dotnet-install: Attempting to download using primary link https://dotnetcli.azureedge.net/dotnet/Sdk/6.0.100/dotnet-sdk-6.0.100-linux-x64.tar.gz
+dotnet-install: Extracting zip from https://dotnetcli.azureedge.net/dotnet/Sdk/6.0.100/dotnet-sdk-6.0.100-linux-x64.tar.gz
+dotnet-install: Adding to current process PATH: `/iot/.dotnet`. Note: This change will be visible only when sourcing script.
+dotnet-install: Installation finished successfully.
+  Determining projects to restore...
+  Restored /iot/eng/common/internal/Tools.csproj (in 5 ms).
+  Determining projects to restore...
+```
+
+There are significant gaps with this approach:
+
+- Only works well in the context of running repo scripts.
+- Doesn't work well for interactive development with `dotnet` CLI, unless you add the private install root (`/iot/.dotnet` in the example) to the `PATH`. That works but isn't convenient for most users.
+- If the global .NET is used accidentally (via the `PATH`), the dev either gets an error due a mismatch with a [repo-resident `global.json` file](https://github.com/dotnet/iot/blob/main/global.json) or it matches "enough" and works but isn't actually the right one and may have slightly different behavior. This may cause the dev to do one of several things, including installing the matching private version globally or editing `global.json`, both of which are not the best practice and a drain on productivity.
+- Works poorly with Visual Studio or Visual Studio Code, where relying on an ephemeral `PATH` setting is awkward.
+- [Multi-level lookup](https://github.com/dotnet/runtime/blob/main/docs/design/features/multilevel-sharedfx-lookup.md) causes a global SDK or runtime to be used, even when you think you are following the best practice to use a private one.
+
+Ideally, there would be a way to lock a repo to a .NET version and location, using a setting similar to `DOTNET_ROOT`. However, environment variables (ENVs) are a poor choice since they are not convenient for using a solution or project file as-is and don't compose nicely with `PATH`. We need an experience that works via a sweet-spot combination of configuration and convention.
+
+## Intuitive version selection
+
+.NET version selection includes cases of being both overly strict and overly lax. Version selection should have a baseline intutitive behavior and provide (as appropriate) additional experiences to meet niche needs.
+
+The most problematic behaviors today are:
+
+- [Multi-level lookup](https://github.com/dotnet/sdk/issues/12353) can result in the global install location being used when a private location is intended.
+- `global.json` defaults to no roll-forward by default, for the specified SDK version. This results in very fragile environments, by default. Instead, the version specified should be a version floor by default.
```

Comment:
I believe we actually rollforward the patch version today (according to the docs).  Are you suggesting a behavior that's not actually a rollforward behavior (because minimum is different than rollforward)? Is rollforward=latestfeature the preferred option or would you want to allow across major band rolling?

Reply:
You are right. I realized this was wrong while I was writing the text and then didn't catch all the places. Thanks.